test/e2e/user_workload_monitoring_test.go

```diff
@@ -532,6 +558,18 @@ func assertUserWorkloadMetrics(t *testing.T) {
 			return fmt.Errorf("expected count of recording rule from user application to be equal 1 but got %v", i)
 		},
 	)
+
+	// Assert that recording rule is in thanos querier and we get it
+	// via user workload prometheus.
+	f.ThanosQuerierClient.WaitForQueryReturn(
+		t, 10*time.Minute, `version:blah:leaf:count{prometheus_replica="prometheus-user-workload-0"}`,
+		func(i int) error {
+			if i == 1 {
+				return nil
+			}
+			return fmt.Errorf("expected count of recording rule from user application to be equal 1 but got %v", i)
+		},
+	)
```

Comment:
we should also add a test that asserts that `version:blah:leaf:count{prometheus_replica="prometheus-user-workload-0"}` does not return any value.